client/src/redux/selectors.js

```diff
@@ -133,11 +134,26 @@ export const completedDailyCodingChallengesIdsSelector = createSelector(
 );
 
 export const completionStateSelector = createSelector(
-  [allChallengesInfoSelector, completedChallengesIdsSelector],
-  (allChallengesInfo, completedChallengesIds) => {
-    const chapters = superBlockStructure.chapters;
+  [
+    allChallengesInfoSelector,
+    completedChallengesIdsSelector,
+    superBlockStructuresSelector,
+    state => state.challenge.challengeMeta
+  ],
+  (
+    allChallengesInfo,
+    completedChallengesIds,
+    superBlockStructures,
+    challengeMeta
+  ) => {
+    if (!chapterBasedSuperBlocks.includes(challengeMeta.superBlock)) return [];
+
     const { challengeNodes } = allChallengesInfo;
 
+    const structure = superBlockStructures[challengeMeta.superBlock];
+
+    const chapters = structure.chapters;
```

Comment:
```suggestion
    const chapters = structure.chapters ?? [];
```
This might be a bit too cute, but this results in the selector returning `[]`

If it's not clear, we could just do `if (isEmpty(chapters)) return [];`